src/main/kotlin/com/coder/gateway/CoderRemoteConnectionHandle.kt

```diff
@@ -412,18 +413,16 @@ class CoderRemoteConnectionHandle {
     ) {
         if (setupCommand.isNotBlank()) {
             indicator.text = "Running setup command..."
-            try {
-                exec(workspace, setupCommand)
-            } catch (ex: Exception) {
-                if (!ignoreSetupFailure) {
-                    throw ex
-                }
-            }
+            processSetupCommand(
+                { exec(workspace, setupCommand) },
```

Comment:
minor: lambda/function parameters are usually declared as last parameter in the function (and only when it's possible). Trailing functions can then be called  in a nicer way:

```
processSetupCommand(ignoreSetupFailure) { 
    exec(workspace, setupCommand)
}
```